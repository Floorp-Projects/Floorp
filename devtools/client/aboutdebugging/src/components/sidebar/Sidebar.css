/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

.sidebar {
  display: grid;
  grid-template-rows: auto auto;
}

.sidebar__label {
  color: var(--secondary-text-color);
  display: block;
  padding: 12px 0;
  text-align: center;
  font-size: var(--message-font-size);
}

.sidebar__adb-status {
  margin-block-end: calc(var(--base-unit) * 2);
}

.sidebar__refresh-usb {
  text-align: center;
}

.sidebar__footer {
  align-self: flex-end;
}

.sidebar__footer__support-help {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  column-gap: calc(var(--base-unit) * 4);
  height: 100%;
}

.sidebar__footer__icon {
  width: calc(var(--base-unit) * 4);
  height: calc(var(--base-unit) * 4);
  -moz-context-properties: fill;
  fill: currentColor;
}
