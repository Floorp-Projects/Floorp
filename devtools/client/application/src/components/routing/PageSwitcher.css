/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */


/*
 * Page container for worker + manifest views
 */

.app-page {
  padding: calc(var(--base-unit) * 3) calc(var(--base-unit) * 8);
  user-select: none;
  overflow-y: auto;
}

.app-page--empty {
  display: grid;
  align-items: center;
  justify-content: center;
  font-size: var(--body-10-font-size);
  color: var(--theme-toolbar-color);
}

.app-page__title {
  font-size: var(--title-20-font-size);
  font-weight: var(--title-20-font-weight);
  margin: 0;
}

.app-page__icon-container {
  display: grid;
  grid-template-columns: auto 1fr;
  grid-column-gap: calc(var(--base-unit) * 4);
}

.app-page__icon {
  width: calc(var(--base-unit) * 10);
  height: calc(var(--base-unit) * 10);

  fill: var(--dimmed-icon-color);
  -moz-context-properties: fill;

  /* alignment fix for text to compensate for low baseline */
  margin-block-start: var(--base-unit);
}
