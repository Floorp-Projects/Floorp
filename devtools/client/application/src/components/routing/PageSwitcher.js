/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const { connect } = require("devtools/client/shared/vendor/react-redux");

const { PAGE_TYPES } = require("devtools/client/application/src/constants");
const Types = require("devtools/client/application/src/types/index");

const ManifestPage = createFactory(
  require("devtools/client/application/src/components/manifest/ManifestPage")
);
const WorkersPage = createFactory(
  require("devtools/client/application/src/components/service-workers/WorkersPage")
);

class PageSwitcher extends PureComponent {
  static get propTypes() {
    return {
      page: Types.page.isRequired,
    };
  }

  render() {
    let component = null;

    switch (this.props.page) {
      case PAGE_TYPES.MANIFEST:
        component = ManifestPage({});
        break;
      case PAGE_TYPES.SERVICE_WORKERS:
        component = WorkersPage({});
        break;
      default:
        console.error("Unknown path. Can not direct to a page.");
        return null;
    }

    return component;
  }
}

function mapStateToProps(state) {
  return {
    page: state.ui.selectedPage,
  };
}

module.exports = connect(mapStateToProps)(PageSwitcher);
