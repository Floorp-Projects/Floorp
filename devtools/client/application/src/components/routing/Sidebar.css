/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */


/*
 * Sidebar list container
 */
.sidebar {
	background-color: var(--bg-color);
}

/* vertical layout -> the sidebar is the first row */
@media(max-width: 700px) {
	.sidebar {
		border-block-end: 1px solid var(--separator-color);
	}
}

/* wide layout -> the sidebar occupies a whole column on the side */
@media(min-width: 701px) {
	.sidebar {
		min-height: 100vh;
		border-inline-end: 1px solid var(--separator-color);
	}
}

.sidebar__list {
	list-style: none;
	padding: 0;
	font-size: var(--body-10-font-size);
	font-weight: var(--body-10-font-weight);
}
