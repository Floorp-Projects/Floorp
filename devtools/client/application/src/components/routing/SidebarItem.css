/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */


/*
 * Sidebar list items
 */

.sidebar-item {
  display: grid;
  grid-template-columns: auto 1fr;
  grid-gap: var(--base-unit);
  padding: calc(var(--base-unit)) calc(var(--base-unit) * 6);
  user-select: none;
  cursor: pointer;
}

.sidebar-item--selected {
	background-color: var(--theme-selection-background);
	color: var(--theme-selection-color);
}

.sidebar-item:not(.sidebar-item--selected):hover {
	background-color: var(--highlight-color);
}

.sidebar-item__icon {
	height: calc(var(--base-unit) * 4);
	width: calc(var(--base-unit) * 4);
	-moz-context-properties: fill;
	fill: currentColor;
}
