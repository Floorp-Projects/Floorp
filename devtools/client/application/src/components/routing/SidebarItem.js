/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const {
  img,
  li,
  span,
} = require("devtools/client/shared/vendor/react-dom-factories");

const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Actions = require("devtools/client/application/src/actions/index");
const { connect } = require("devtools/client/shared/vendor/react-redux");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const Localized = createFactory(FluentReact.Localized);

const { PAGE_TYPES } = require("devtools/client/application/src/constants");
const Types = require("devtools/client/application/src/types/index");

const ICONS = {
  [PAGE_TYPES.MANIFEST]:
    "chrome://devtools/skin/images/application-manifest.svg",
  [PAGE_TYPES.SERVICE_WORKERS]:
    "chrome://devtools/skin/images/debugging-workers.svg",
};

const LOCALIZATION_IDS = {
  [PAGE_TYPES.MANIFEST]: "sidebar-item-manifest",
  [PAGE_TYPES.SERVICE_WORKERS]: "sidebar-item-service-workers",
};

class SidebarItem extends PureComponent {
  static get propTypes() {
    return {
      page: Types.page.isRequired,
      isSelected: PropTypes.bool.isRequired,
      // this prop is automatically injected via connect
      dispatch: PropTypes.func.isRequired,
    };
  }

  render() {
    const { isSelected, page } = this.props;

    return li(
      {
        className: `sidebar-item js-sidebar-${page} ${
          isSelected ? "sidebar-item--selected" : ""
        }`,
        onClick: () => {
          const { dispatch } = this.props;
          dispatch(Actions.updateSelectedPage(page));
        },
        role: "link",
      },
      Localized(
        {
          id: LOCALIZATION_IDS[page],
          attrs: {
            alt: true,
            title: true,
          },
        },
        img({
          src: ICONS[page],
          className: "sidebar-item__icon",
        })
      ),
      Localized(
        {
          id: LOCALIZATION_IDS[page],
          attrs: {
            title: true,
          },
        },
        span({ className: "devtools-ellipsis-text" })
      )
    );
  }
}

const mapDispatchToProps = dispatch => ({ dispatch });
module.exports = connect(mapDispatchToProps)(SidebarItem);
