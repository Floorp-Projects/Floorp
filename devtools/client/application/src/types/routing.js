/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const PropTypes = require("devtools/client/shared/vendor/react-prop-types");
const { PAGE_TYPES } = require("devtools/client/application/src/constants");

const page = PropTypes.oneOf(Object.values(PAGE_TYPES));

module.exports = {
  page,
};
