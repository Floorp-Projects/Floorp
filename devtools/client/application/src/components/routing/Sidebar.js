/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const {
  aside,
  ul,
} = require("devtools/client/shared/vendor/react-dom-factories");

const { connect } = require("devtools/client/shared/vendor/react-redux");

const SidebarItem = createFactory(
  require("devtools/client/application/src/components/routing/SidebarItem")
);

const Types = require("devtools/client/application/src/types/index");
const { PAGE_TYPES } = require("devtools/client/application/src/constants");

class Sidebar extends PureComponent {
  static get propTypes() {
    return {
      // this prop is automatically injected via connect
      selectedPage: Types.page.isRequired,
    };
  }

  render() {
    const navItems = [PAGE_TYPES.SERVICE_WORKERS, PAGE_TYPES.MANIFEST];

    const isSelected = page => {
      return page === this.props.selectedPage;
    };

    return aside(
      {
        className: "sidebar js-sidebar",
      },
      ul(
        {
          className: "sidebar__list",
        },
        navItems.map(page => {
          return SidebarItem({
            page: page,
            key: `sidebar-item-${page}`,
            isSelected: isSelected(page),
          });
        })
      )
    );
  }
}

function mapStateToProps(state) {
  return {
    selectedPage: state.ui.selectedPage,
  };
}

module.exports = connect(mapStateToProps)(Sidebar);
