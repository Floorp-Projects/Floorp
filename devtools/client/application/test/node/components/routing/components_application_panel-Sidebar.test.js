/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

// Import libs
const { shallow } = require("enzyme");
const { createFactory } = require("react");

const { setupStore } = require("devtools/client/application/test/node/helpers");

const { PAGE_TYPES } = require("devtools/client/application/src/constants");

const Sidebar = createFactory(
  require("devtools/client/application/src/components/routing/Sidebar")
);

/**
 * Test for Sidebar.js component
 */

describe("Sidebar", () => {
  function buildStoreWithSelectedPage(selectedPage) {
    return setupStore({
      ui: {
        selectedPage,
      },
    });
  }
  it("renders the expected snapshot when the manifest page is selected", () => {
    const store = buildStoreWithSelectedPage(PAGE_TYPES.MANIFEST);
    const wrapper = shallow(Sidebar({ store })).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the expected snapshot when the service workers page is selected", () => {
    const store = buildStoreWithSelectedPage(PAGE_TYPES.SERVICE_WORKERS);
    const wrapper = shallow(Sidebar({ store })).dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the expected snapshot when no page is selected", () => {
    const store = buildStoreWithSelectedPage();
    const wrapper = shallow(Sidebar({ store })).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
