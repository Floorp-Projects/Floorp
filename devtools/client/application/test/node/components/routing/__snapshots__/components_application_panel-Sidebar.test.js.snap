// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders the expected snapshot when no page is selected 1`] = `
<aside
  className="sidebar js-sidebar"
>
  <ul
    className="sidebar__list"
  >
    <Connect(SidebarItem)
      isSelected={false}
      key="sidebar-item-service-workers"
      page="service-workers"
    />
    <Connect(SidebarItem)
      isSelected={false}
      key="sidebar-item-manifest"
      page="manifest"
    />
  </ul>
</aside>
`;

exports[`Sidebar renders the expected snapshot when the manifest page is selected 1`] = `
<aside
  className="sidebar js-sidebar"
>
  <ul
    className="sidebar__list"
  >
    <Connect(SidebarItem)
      isSelected={false}
      key="sidebar-item-service-workers"
      page="service-workers"
    />
    <Connect(SidebarItem)
      isSelected={true}
      key="sidebar-item-manifest"
      page="manifest"
    />
  </ul>
</aside>
`;

exports[`Sidebar renders the expected snapshot when the service workers page is selected 1`] = `
<aside
  className="sidebar js-sidebar"
>
  <ul
    className="sidebar__list"
  >
    <Connect(SidebarItem)
      isSelected={true}
      key="sidebar-item-service-workers"
      page="service-workers"
    />
    <Connect(SidebarItem)
      isSelected={false}
      key="sidebar-item-manifest"
      page="manifest"
    />
  </ul>
</aside>
`;
